### PR TITLE
Dockerfile: copy only CA certificate bundle

### PR DESCRIPTION
**Pull request by rockdrilla**
### 1. Why is this pull request needed and what does it do?

Final container image receives a lot of (dangling) symlinks in /etc/ssl/certs/ while only one (regular) file is required.

### 2. Which issues (if any) are related?

None.

### 3. Which documentation changes (if any) need to be made?

None.

### 4. Does this introduce a backward incompatible change or deprecation?

No.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,10 @@
 FROM debian:stable-slim
 
-RUN apt-get update && apt-get -uy upgrade
-RUN apt-get -y install ca-certificates && update-ca-certificates
+RUN apt-get -qq update && apt-get -yyqq install ca-certificates && apt-get clean
```
Comment (chrisohaver) [inline Dockerfile]: What is the reason for this change?  I see it removes `apt-get upgrade` adds `apt-get clean`.

Comment (rockdrilla) [reply Dockerfile]: apt-get clean - clean up layer (speeds up build)
apt-get upgrade - doesn't affect layer product (i.e. /etc/ssl/certs/ca-certificates.crt) in any way

**Comment by rockdrilla** [comment]
few words about "apt-get upgrade" removal:

$ bud Dockerfile -t coredns-stable-slim-no-upgrade
```
[1/2] STEP 1/2: FROM debian:stable-slim
[1/2] STEP 2/2: RUN apt-get -qq update && apt-get -yyqq install ca-certificates && apt-get clean && sha256sum /etc/ssl/certs/ca-certificates.crt
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package openssl.
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../openssl_1.1.1n-0+deb11u3_amd64.deb ...
Unpacking openssl (1.1.1n-0+deb11u3) ...
Selecting previously unselected package ca-certificates.
Preparing to unpack .../ca-certificates_20210119_all.deb ...
Unpacking ca-certificates (20210119) ...
Setting up openssl (1.1.1n-0+deb11u3) ...
Setting up ca-certificates (20210119) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
Updating certificates in /etc/ssl/certs...
129 added, 0 removed; done.
Processing triggers for ca-certificates (20210119) ...
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
333e611c066ee34adefa78c4f1f3fa710a7f50ffa5fbd2fd53987b2336feaa13  /etc/ssl/certs/ca-certificates.crt
[2/2] STEP 1/5: FROM scratch
[2/2] STEP 2/5: COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
[2/2] STEP 3/5: ADD coredns /coredns
[2/2] STEP 4/5: EXPOSE 53 53/udp
[2/2] STEP 5/5: ENTRYPOINT ["/coredns"]
[2/2] COMMIT coredns-stable-slim-no-upgrade
Getting image source signatures
Copying blob b08dee67e7f6 done
Copying config 4348066e61 done
Writing manifest to image destination
Storing signatures
--> 4348066e613
Successfully tagged localhost/coredns-stable-slim-no-upgrade:latest
4348066e613b09f876891045ea021e2e42d9eca979ad99559ad66f00c5d9946d
```

$ bud Dockerfile -t coredns-old-but-stable-slim-no-upgrade --from debian:bullseye-20210902-slim
```
[1/2] STEP 1/2: FROM debian:bullseye-20210902-slim
[1/2] STEP 2/2: RUN apt-get -qq update && apt-get -yyqq install ca-certificates && apt-get clean && sha256sum /etc/ssl/certs/ca-certificates.crt
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package openssl.
(Reading database ... 6653 files and directories currently installed.)
Preparing to unpack .../openssl_1.1.1n-0+deb11u3_amd64.deb ...
Unpacking openssl (1.1.1n-0+deb11u3) ...
Selecting previously unselected package ca-certificates.
Preparing to unpack .../ca-certificates_20210119_all.deb ...
Unpacking ca-certificates (20210119) ...
Setting up openssl (1.1.1n-0+deb11u3) ...
Setting up ca-certificates (20210119) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
Updating certificates in /etc/ssl/certs...
129 added, 0 removed; done.
Processing triggers for ca-certificates (20210119) ...
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
333e611c066ee34adefa78c4f1f3fa710a7f50ffa5fbd2fd53987b2336feaa13  /etc/ssl/certs/ca-certificates.crt
[2/2] STEP 1/5: FROM scratch
[2/2] STEP 2/5: COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
[2/2] STEP 3/5: ADD coredns /coredns
[2/2] STEP 4/5: EXPOSE 53 53/udp
[2/2] STEP 5/5: ENTRYPOINT ["/coredns"]
[2/2] COMMIT coredns-old-but-stable-slim-no-upgrade
Getting image source signatures
Copying blob 4e0970742abe done
Copying config b217d343a0 done
Writing manifest to image destination
Storing signatures
--> b217d343a06
Successfully tagged localhost/coredns-old-but-stable-slim-no-upgrade:latest
b217d343a06ce4562137aa059bbf116f0a7abad54fc7f790502337d4cc6e7e8f
```

$ bud Dockerfile -t coredns-stable-slim-with-upgrade
```
[1/2] STEP 1/2: FROM debian:stable-slim
[1/2] STEP 2/2: RUN apt-get -qq update && apt-get -yyqq upgrade && apt-get -yyqq install ca-certificates && apt-get clean && sha256sum /etc/ssl/certs/ca-certificates.crt
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libgnutls30_3.7.1-5+deb11u2_amd64.deb ...
Unpacking libgnutls30:amd64 (3.7.1-5+deb11u2) over (3.7.1-5+deb11u1) ...
Setting up libgnutls30:amd64 (3.7.1-5+deb11u2) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libtirpc-common_1.3.1-1+deb11u1_all.deb ...
Unpacking libtirpc-common (1.3.1-1+deb11u1) over (1.3.1-1) ...
Setting up libtirpc-common (1.3.1-1+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libtirpc3_1.3.1-1+deb11u1_amd64.deb ...
Unpacking libtirpc3:amd64 (1.3.1-1+deb11u1) over (1.3.1-1) ...
Setting up libtirpc3:amd64 (1.3.1-1+deb11u1) ...
Processing triggers for libc-bin (2.31-13+deb11u3) ...
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package openssl.
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../openssl_1.1.1n-0+deb11u3_amd64.deb ...
Unpacking openssl (1.1.1n-0+deb11u3) ...
Selecting previously unselected package ca-certificates.
Preparing to unpack .../ca-certificates_20210119_all.deb ...
Unpacking ca-certificates (20210119) ...
Setting up openssl (1.1.1n-0+deb11u3) ...
Setting up ca-certificates (20210119) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
Updating certificates in /etc/ssl/certs...
129 added, 0 removed; done.
Processing triggers for ca-certificates (20210119) ...
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
333e611c066ee34adefa78c4f1f3fa710a7f50ffa5fbd2fd53987b2336feaa13  /etc/ssl/certs/ca-certificates.crt
[2/2] STEP 1/5: FROM scratch
[2/2] STEP 2/5: COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
[2/2] STEP 3/5: ADD coredns /coredns
[2/2] STEP 4/5: EXPOSE 53 53/udp
[2/2] STEP 5/5: ENTRYPOINT ["/coredns"]
[2/2] COMMIT coredns-stable-slim-with-upgrade
Getting image source signatures
Copying blob 540ec085fb1e done
Copying config 7b8f630c7e done
Writing manifest to image destination
Storing signatures
--> 7b8f630c7e8
Successfully tagged localhost/coredns-stable-slim-with-upgrade:latest
7b8f630c7e8548952685f95f2b4a98f4c8690f64e14e3f5e469a871d1b7e2f23
```

$ bud Dockerfile -t coredns-old-but-stable-slim-with-upgrade --from debian:bullseye-20210902-slim
```
[1/2] STEP 1/2: FROM debian:bullseye-20210902-slim
[1/2] STEP 2/2: RUN apt-get -qq update && apt-get -yyqq upgrade && apt-get -yyqq install ca-certificates && apt-get clean && sha256sum /etc/ssl/certs/ca-certificates.crt
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 6653 files and directories currently installed.)
Preparing to unpack .../base-files_11.1+deb11u4_amd64.deb ...
Unpacking base-files (11.1+deb11u4) over (11.1) ...
Setting up base-files (11.1+deb11u4) ...
Installing new version of config file /etc/debian_version ...
(Reading database ... 6653 files and directories currently installed.)
Preparing to unpack .../bash_5.1-2+deb11u1_amd64.deb ...
update-alternatives: warning: alternative /usr/share/man/man7/bash-builtins.7.gz (part of link group builtins.7.gz) doesn't exist; removing from list of alternatives
update-alternatives: warning: /etc/alternatives/builtins.7.gz is dangling; it will be updated with best choice
Unpacking bash (5.1-2+deb11u1) over (5.1-2+b3) ...
Setting up bash (5.1-2+deb11u1) ...
update-alternatives: error: alternative path /usr/share/man/man7/bash-builtins.7.gz doesn't exist
(Reading database ... 6652 files and directories currently installed.)
Preparing to unpack .../bsdutils_1%3a2.36.1-8+deb11u1_amd64.deb ...
Unpacking bsdutils (1:2.36.1-8+deb11u1) over (1:2.36.1-8) ...
Setting up bsdutils (1:2.36.1-8+deb11u1) ...
(Reading database ... 6652 files and directories currently installed.)
Preparing to unpack .../dpkg_1.20.11_amd64.deb ...
Unpacking dpkg (1.20.11) over (1.20.9) ...
Setting up dpkg (1.20.11) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../gzip_1.10-4+deb11u1_amd64.deb ...
Unpacking gzip (1.10-4+deb11u1) over (1.10-4) ...
Setting up gzip (1.10-4+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../perl-base_5.32.1-4+deb11u2_amd64.deb ...
Unpacking perl-base (5.32.1-4+deb11u2) over (5.32.1-4+deb11u1) ...
Setting up perl-base (5.32.1-4+deb11u2) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../util-linux_2.36.1-8+deb11u1_amd64.deb ...
Unpacking util-linux (2.36.1-8+deb11u1) over (2.36.1-8) ...
Setting up util-linux (2.36.1-8+deb11u1) ...
update-alternatives: warning: forcing reinstallation of alternative /bin/more because link group pager is broken
update-alternatives: warning: skip creation of /usr/share/man/man1/pager.1.gz because associated file /usr/share/man/man1/more.1.gz (of link group pager) doesn't exist
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libc6_2.31-13+deb11u3_amd64.deb ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
Unpacking libc6:amd64 (2.31-13+deb11u3) over (2.31-13) ...
Setting up libc6:amd64 (2.31-13+deb11u3) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libc-bin_2.31-13+deb11u3_amd64.deb ...
Unpacking libc-bin (2.31-13+deb11u3) over (2.31-13) ...
Setting up libc-bin (2.31-13+deb11u3) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../sysvinit-utils_2.96-7+deb11u1_amd64.deb ...
Unpacking sysvinit-utils (2.96-7+deb11u1) over (2.96-7) ...
Setting up sysvinit-utils (2.96-7+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libpam0g_1.4.0-9+deb11u1_amd64.deb ...
Unpacking libpam0g:amd64 (1.4.0-9+deb11u1) over (1.4.0-9) ...
Setting up libpam0g:amd64 (1.4.0-9+deb11u1) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libpam-modules-bin_1.4.0-9+deb11u1_amd64.deb ...
Unpacking libpam-modules-bin (1.4.0-9+deb11u1) over (1.4.0-9) ...
Setting up libpam-modules-bin (1.4.0-9+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libpam-modules_1.4.0-9+deb11u1_amd64.deb ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
Unpacking libpam-modules:amd64 (1.4.0-9+deb11u1) over (1.4.0-9) ...
Setting up libpam-modules:amd64 (1.4.0-9+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libsystemd0_247.3-7_amd64.deb ...
Unpacking libsystemd0:amd64 (247.3-7) over (247.3-6) ...
Setting up libsystemd0:amd64 (247.3-7) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../mount_2.36.1-8+deb11u1_amd64.deb ...
Unpacking mount (2.36.1-8+deb11u1) over (2.36.1-8) ...
Preparing to unpack .../libpam-runtime_1.4.0-9+deb11u1_all.deb ...
Unpacking libpam-runtime (1.4.0-9+deb11u1) over (1.4.0-9) ...
Setting up libpam-runtime (1.4.0-9+deb11u1) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../zlib1g_1%3a1.2.11.dfsg-2+deb11u1_amd64.deb ...
Unpacking zlib1g:amd64 (1:1.2.11.dfsg-2+deb11u1) over (1:1.2.11.dfsg-2) ...
Setting up zlib1g:amd64 (1:1.2.11.dfsg-2+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../gpgv_2.2.27-2+deb11u2_amd64.deb ...
Unpacking gpgv (2.2.27-2+deb11u2) over (2.2.27-2) ...
Setting up gpgv (2.2.27-2+deb11u2) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libblkid1_2.36.1-8+deb11u1_amd64.deb ...
Unpacking libblkid1:amd64 (2.36.1-8+deb11u1) over (2.36.1-8) ...
Setting up libblkid1:amd64 (2.36.1-8+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libgmp10_2%3a6.2.1+dfsg-1+deb11u1_amd64.deb ...
Unpacking libgmp10:amd64 (2:6.2.1+dfsg-1+deb11u1) over (2:6.2.1+dfsg-1) ...
Setting up libgmp10:amd64 (2:6.2.1+dfsg-1+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libgnutls30_3.7.1-5+deb11u2_amd64.deb ...
Unpacking libgnutls30:amd64 (3.7.1-5+deb11u2) over (3.7.1-5) ...
Setting up libgnutls30:amd64 (3.7.1-5+deb11u2) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libk5crypto3_1.18.3-6+deb11u1_amd64.deb ...
Unpacking libk5crypto3:amd64 (1.18.3-6+deb11u1) over (1.18.3-6) ...
Setting up libk5crypto3:amd64 (1.18.3-6+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libkrb5support0_1.18.3-6+deb11u1_amd64.deb ...
Unpacking libkrb5support0:amd64 (1.18.3-6+deb11u1) over (1.18.3-6) ...
Setting up libkrb5support0:amd64 (1.18.3-6+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libkrb5-3_1.18.3-6+deb11u1_amd64.deb ...
Unpacking libkrb5-3:amd64 (1.18.3-6+deb11u1) over (1.18.3-6) ...
Setting up libkrb5-3:amd64 (1.18.3-6+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libgssapi-krb5-2_1.18.3-6+deb11u1_amd64.deb ...
Unpacking libgssapi-krb5-2:amd64 (1.18.3-6+deb11u1) over (1.18.3-6) ...
Setting up libgssapi-krb5-2:amd64 (1.18.3-6+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libssl1.1_1.1.1n-0+deb11u3_amd64.deb ...
Unpacking libssl1.1:amd64 (1.1.1n-0+deb11u3) over (1.1.1k-1+deb11u1) ...
Setting up libssl1.1:amd64 (1.1.1n-0+deb11u3) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../liblzma5_5.2.5-2.1~deb11u1_amd64.deb ...
Unpacking liblzma5:amd64 (5.2.5-2.1~deb11u1) over (5.2.5-2) ...
Setting up liblzma5:amd64 (5.2.5-2.1~deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libmount1_2.36.1-8+deb11u1_amd64.deb ...
Unpacking libmount1:amd64 (2.36.1-8+deb11u1) over (2.36.1-8) ...
Setting up libmount1:amd64 (2.36.1-8+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libseccomp2_2.5.1-1+deb11u1_amd64.deb ...
Unpacking libseccomp2:amd64 (2.5.1-1+deb11u1) over (2.5.1-1) ...
Setting up libseccomp2:amd64 (2.5.1-1+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libsmartcols1_2.36.1-8+deb11u1_amd64.deb ...
Unpacking libsmartcols1:amd64 (2.36.1-8+deb11u1) over (2.36.1-8) ...
Setting up libsmartcols1:amd64 (2.36.1-8+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libtirpc-common_1.3.1-1+deb11u1_all.deb ...
Unpacking libtirpc-common (1.3.1-1+deb11u1) over (1.3.1-1) ...
Setting up libtirpc-common (1.3.1-1+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libtirpc3_1.3.1-1+deb11u1_amd64.deb ...
Unpacking libtirpc3:amd64 (1.3.1-1+deb11u1) over (1.3.1-1) ...
Setting up libtirpc3:amd64 (1.3.1-1+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libudev1_247.3-7_amd64.deb ...
Unpacking libudev1:amd64 (247.3-7) over (247.3-6) ...
Setting up libudev1:amd64 (247.3-7) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../libuuid1_2.36.1-8+deb11u1_amd64.deb ...
Unpacking libuuid1:amd64 (2.36.1-8+deb11u1) over (2.36.1-8) ...
Setting up libuuid1:amd64 (2.36.1-8+deb11u1) ...
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../tzdata_2021a-1+deb11u4_all.deb ...
Unpacking tzdata (2021a-1+deb11u4) over (2021a-1) ...
Setting up tzdata (2021a-1+deb11u4) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype

Current default time zone: 'Etc/UTC'
Local time is now:      Mon Aug 15 14:29:50 UTC 2022.
Universal Time is now:  Mon Aug 15 14:29:50 UTC 2022.
Run 'dpkg-reconfigure tzdata' if you wish to change it.

Setting up mount (2.36.1-8+deb11u1) ...
Processing triggers for libc-bin (2.31-13+deb11u3) ...
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package openssl.
(Reading database ... 6661 files and directories currently installed.)
Preparing to unpack .../openssl_1.1.1n-0+deb11u3_amd64.deb ...
Unpacking openssl (1.1.1n-0+deb11u3) ...
Selecting previously unselected package ca-certificates.
Preparing to unpack .../ca-certificates_20210119_all.deb ...
Unpacking ca-certificates (20210119) ...
Setting up openssl (1.1.1n-0+deb11u3) ...
Setting up ca-certificates (20210119) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
Updating certificates in /etc/ssl/certs...
129 added, 0 removed; done.
Processing triggers for ca-certificates (20210119) ...
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
333e611c066ee34adefa78c4f1f3fa710a7f50ffa5fbd2fd53987b2336feaa13  /etc/ssl/certs/ca-certificates.crt
[2/2] STEP 1/5: FROM scratch
[2/2] STEP 2/5: COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
[2/2] STEP 3/5: ADD coredns /coredns
[2/2] STEP 4/5: EXPOSE 53 53/udp
[2/2] STEP 5/5: ENTRYPOINT ["/coredns"]
[2/2] COMMIT coredns-old-but-stable-slim-with-upgrade
Getting image source signatures
Copying blob 1da435bafbb8 done
Copying config f1f1b47f30 done
Writing manifest to image destination
Storing signatures
--> f1f1b47f30f
Successfully tagged localhost/coredns-old-but-stable-slim-with-upgrade:latest
f1f1b47f30f4bd138643e24eb608f51761824276752b806e4fcf83afb275a33d
```

As one may see, /etc/ssl/certs/ca-certificates.crt is identical in all four cases (check sha256sum - 333e611c066ee34adefa78c4f1f3fa710a7f50ffa5fbd2fd53987b2336feaa13).

**Comment by yongtang** [comment]
@rockdrilla I think we want to keep `apt-get upgrade` to patch potential security fixes.

**Comment by rockdrilla** [comment]
@chrisohaver, @yongtang, "apt-get upgrade" is back.

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,12 @@
 FROM debian:stable-slim
+SHELL [ "/bin/sh", "-ec" ]
 
-RUN apt-get update && apt-get -uy upgrade
-RUN apt-get -y install ca-certificates && update-ca-certificates
+ENV DEBCONF_NONINTERACTIVE_SEEN=true DEBIAN_FRONTEND=noninteractive DEBIAN_PRIORITY=critical TERM=linux
```
Comment (chrisohaver) [inline Dockerfile]: Why are these required?

**Comment by yongtang** [comment]
@rockdrilla  I think `ENV DEBCONF_NONINTERACTIVE_SEEN=true DEBIAN_FRONTEND=noninteractive DEBIAN_PRIORITY=critical TERM=linux` will stick the env change? Can you move this to the line of `RUN` instead? This will only impact at the time of the build.

File: Dockerfile (hunk 2 above)
Comment (rockdrilla) [reply Dockerfile]: These environment variables reduce amount of (useless) information in build logs, mostly debconf tuning.

**Comment by rockdrilla** [comment]
@yongtang, this ENV clause affects only this intermediate layer.

**Comment by rockdrilla** [comment]
I've merged two clauses (ENV, RUN) into one RUN.

File: Dockerfile [hunk 3]
```diff
@@ -1,11 +1,18 @@
 FROM debian:stable-slim
+SHELL [ "/bin/sh", "-ec" ]
 
-RUN apt-get update && apt-get -uy upgrade
-RUN apt-get -y install ca-certificates && update-ca-certificates
```
Comment (chrisohaver) [inline Dockerfile:4]: Is running `update-ca-certificates` no longer necessary?

Comment (rockdrilla) [reply Dockerfile:4]: This is done automatically by postinst script in "ca-certificates" package.

**Review by yongtang — APPROVED** [review]
